jwst/resample/tests/test_resample_step.py

```diff
@@ -127,18 +120,129 @@ def test_spatial_transform_miri():
+        'xsize': 2048,
+        'xstart': 1,
+        'ysize': 2048,
+        'ystart': 1}
+    observation = {
+        'activity_id': '01',
+        'date': '2021-10-25',
+        'exposure_number': '00001',
+        'obs_id': 'V42424001001P0000000001101',
+        'observation_label': 'nircam_ptsrc_only',
+        'observation_number': '001',
+        'program_number': '42424',
+        'sequence_id': '1',
+        'time': '16:58:27.258',
+        'visit_group': '01',
+        'visit_id': '42424001001',
+        'visit_number': '001'}
+    exposure = {
+        'duration': 161.05155,
+        'end_time': 59512.70899968495,
+        'exposure_time': 150.31478,
+        'frame_time': 10.73677,
+        'group_time': 21.47354,
+        'groupgap': 1,
+        'integration_time': 150.31478,
+        'mid_time': 59512.70812980775,
+        'nframes': 1,
+        'ngroups': 7,
+        'nints': 1,
+        'nresets_at_start': 1,
+        'nresets_between_ints': 1,
+        'readpatt': 'BRIGHT1',
+        'sample_time': 10,
+        'start_time': 59512.70725993055,
+        'type': 'NRC_IMAGE'}
+
+    im = ImageModel((2048, 2048))
+    im.data += 5.
+
+    im.meta.wcsinfo = wcsinfo
+    im.meta.instrument = instrument
+    im.meta.observation = observation
+    im.meta.exposure = exposure
+    im.meta.subarray = subarray
+
+    return im
+
+
+def test_nirspec_wcs_roundtrip(nirspec_rate):
+    im = AssignWcsStep.call(nirspec_rate)
+    im = Extract2dStep.call(im)
+    im = ResampleSpecStep.call(im)
+
+    for slit in im.slits:
+        x, y = grid_from_bounding_box(slit.meta.wcs.bounding_box)
+        ra, dec, lam = slit.meta.wcs(x, y)
+        xp, yp = slit.meta.wcs.invert(ra, dec, lam)
+
+        assert_allclose(x, xp, atol=1e-8)
+        assert_allclose(y, yp, atol=1e-8)
+
+
+def test_miri_wcs_roundtrip(miri_rate):
+    im = AssignWcsStep.call(miri_rate)
     im = ResampleSpecStep.call(im)
 
     x, y = grid_from_bounding_box(im.meta.wcs.bounding_box)
     ra, dec, lam = im.meta.wcs(x, y)
     xp, yp = im.meta.wcs.invert(ra, dec, lam)
     assert_allclose(x, xp, atol=1e-8)
     assert_allclose(y, yp, atol=1e-8)
+
+
+@pytest.mark.parametrize("ratio", [0.5, 0.7, 1.0])
+def test_pixel_scale_ratio_spec(miri_rate, ratio):
+    im = AssignWcsStep.call(miri_rate)
+    result1 = ResampleSpecStep.call(im)
+    result2 = ResampleSpecStep.call(im, pixel_scale_ratio=ratio)
+    assert_allclose(np.array(result1.data.shape), np.array(result2.data.shape) * ratio, rtol=1, atol=1)
+
+
+@pytest.mark.parametrize("ratio", [0.5, 0.7, 1.0])
+def test_pixel_scale_ratio_imaging(nircam_rate, ratio):
+    im = AssignWcsStep.call(nircam_rate)
+    result1 = ResampleStep.call(im)
+    result2 = ResampleStep.call(im, pixel_scale_ratio=ratio)
+    assert_allclose(np.array(result1.data.shape), np.array(result2.data.shape) * ratio, rtol=1, atol=1)
+    e1 = 50
+    e2 = int(50 / ratio)
+    assert np.mean(result1.data[e1:-e1,e1:-e1]) == np.mean(result2.data[e2:-e2,e2:-e2])
```

Comment:
This is where I think things are wrong: the flux in each pixel has to be rescaled by `pixel_scale_ratio`. This test shows that the flux of a star (if one were to do photometry on the resampled image) would quadruple. For the very least this is how I would interpret this for HST images. I do not know what if there are any *fundamental* @hbushouse differences between HST and JWST images. To the best of my knowledge, I think JWST images are fundamentally like `FLT * PHOTFLAM`, i.e., instead of counts or count-rates they have units of flux. If so, the test above shows that the flux is not preserved.